books/RayTracingTheRestOfYourLife.html

```diff
@@ -1680,67 +1679,81 @@
             if (!world.hit(r, interval(0.001, infinity), rec))
                 return background;
 
-
-    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++ highlight
-            color albedo;
             ray scattered;
-            double pdf;
+            color attenuation;
             color emitted = rec.mat->emitted(rec.u, rec.v, rec.p);
 
-            if (!rec.mat->scatter(r, rec, albedo, scattered, pdf))
+            if (!rec.mat->scatter(r, rec, attenuation, scattered))
                 return emitted;
 
-            return emitted
-                 + albedo * rec.mat->scattering_pdf(r, rec, scattered)
-                          * ray_color(scattered, depth-1) / pdf;
+
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++ highlight
+            double scattering_pdf = rec.mat->scattering_pdf(r, rec, scattered);
+            double pdf = scattering_pdf;
```

Comment:
I don't think this is right. If it was, then `scattering_pdf` and `pdf` below would just cancel out.

Ah, never mind. You're doing this for exposition.